docs/deployment.md

```diff
@@ -0,0 +1,228 @@
+
+This is necessary for the server to create the genesis block.
+
+The `genesis root` hash is added in the `./genesis.log` and `./deployed_contracts.log` files.
+
+The `genesis root` hash is also updated in the environment variables and the `toml` files from the current
+`env profile`.
+
+#### Existing deployment
+
+In an existing deployment, the genesis block already exists therefore we just need to connect the server to the
+database.
+
+The environment variables from the `env profile` needs to match with the genesis root hash from the current database.
+
+### Contracts
+
+#### New deployments
+
+During the new deployment process, the contracts are built, deployed and verified with etherscan.
+
+The `contracts` hash are added in the `./deployed_contracts.log` file and we can find extra information in the
+`./deploy.log` file.
+
+The `contracts` hash is also updated in the environment variables and the `toml` files from the current `env profile`.
+
+#### Existing deployment
+
+In an existing deployment, the contracts were already deployed therefore we just need to run the server.
+
+The environment variables from the `env profile` needs to match with the addresses from the `server_config` table from
+the database.
+
+## Prover
+
+To use the `dummy-prover` we need to enable it, prior enabling it, the contracts must be already deployed. During the
+contracts deployment, the `genesis.log` file is generated; this file includes the variable `CONTRACTS_GENESIS_ROOT` and
+it's required for the dummy-prover to be enabled. To enable the `dummy-prover` on docker:
+
+```
+zk dummy-prover enable
+```
+
+Once the `dummy-prover` is enabled, it can be run with the following command:
+
+```
+zk dummy-prover run
+```
+
+## Miscellaneous
+
+### Local environment
+
+<!-- markdownlint-disable MD029-->
+
+We are going to deploy a local environment using the deploy script. To do it; we need to follow the next steps. All the
+information necessary to understand the steps are described above:
+
+1. Create the `dev` environment.
+
+```
+ENV_OVERRIDE=deploy zk init --no-sdk --with-docker
+```
+
+2. Enable the `dummy-prover` for local development.
+
+```
+ENV_OVERRIDE=deploy zk dummy-prover --with-docker enable
+```
+
+3. Start running the `server`.
+
+```
+ENV_OVERRIDE=deploy zk server --with-docker
+```
+
+4. Start running the `dummy-prover`.
+
+```
+ENV_OVERRIDE=deploy zk dummy-prover --with-docker run
+```
+
+<!-- markdownlint-enable MD029-->
+
+### Rollup environment
+
+A docker image was created to facilitate a way of preparing the environment from a docker container and avoid having all
+the dependencies installed in the local host.
+
+<!-- markdownlint-disable MD029-->
+
+1. Create the `dev` environment.
+
+```
+zk config compile dev
+```
+
+2. Initialize docker containers with `rollup` environent, `RSKj` node`postgres` database for local development.
+
+```
+docker-compose -f docker-compose.deploy.yml up -d rskj postgres rollup
```

Comment:
@franciscotobar  This command raise an error because in `docker-compose.deploy.yml`, we don't expose any port for the postgres service

Reply:
updated